NewDawn/NewDawn/ProfileViewController/Profile_Smoke.swift

```diff
@@ -8,73 +8,104 @@
 
 import UIKit
 
+
 class Profile_Smoke: UIViewController {
-    var visible = false
-    var smoke = false
-    var socially = false
-    var frequent = false
+    let NOSMOKE = "no"
+    let SOCIAL_SMOKE = "social"
+    let FREQUENT_SMOKE = "frequent"
+    let VISIBLE = "smoke_visible"
+    var visible_state = false
+    var smoke_pref: String? = nil
     
     @IBOutlet weak var sociallyButton: UIButton!
     @IBOutlet weak var frequentlyButton: UIButton!
     @IBOutlet weak var noSmokeButton: UIButton!
     @IBOutlet weak var visibleButton: UIButton!
     
     
+    func loadStoredFields() {
+        // Select the buttons if a user has already done so
+        if let smoke = localReadKeyValue(key: "smoke") as? String{
```

Comment:
Let's also make "smoke" a constant since it is used in two places (one here and another below)